src/v/cluster/self_test_rpc_types.h

```diff
@@ -233,7 +233,7 @@ struct cloudcheck_opts
 
 struct self_test_result
   : serde::
-      envelope<self_test_result, serde::version<0>, serde::compat_version<0>> {
+      envelope<self_test_result, serde::version<1>, serde::compat_version<1>> {
```

Comment:
bumping the compat version here seems a little sketchy in that this means older nodes can't read this message. but it also seems like they could if we appended the fields? anyway, i would expect that there woudl be a comment and explanation and maybe a test anytime we see compat version being bumped.

Reply:
Yeah it does look like we could append the new fields and not bump compat_version at all... presumably we can still wind this back without violating policy, but I'm not confident to say how disruptive that would be downstream (where rc3 is running currently, who is running it, etc) vs how likely this is to cause problems if we leave it as is.